examples/CMakeLists.txt

```diff
@@ -29,7 +29,7 @@ endif()
 
 # Add Subdirectories
 add_subdirectory (cc)
-add_subdirectory (cuda)
+add_subdirectory (gpu)
```

Comment:
I propose we build out terminology around `device` and `host` rather than `gpu` and `cpu`, to make other types of devices (fpga?) covered

Reply:
I can redo the PR by replacing suffix "gpu" with siffix "device", e.g. examples/gpu -> examples/device, etc. Should I also change the suffix "cpu" to suffix "host" in file names?